src/libnnstreamer-edge/nnstreamer-edge-queue.c

```diff
@@ -177,12 +199,17 @@ nns_edge_queue_push (nns_edge_queue_h handle, void *data,
   qdata->destroy = destroy;
 
   nns_edge_lock (q);
-  if (!q->head)
-    q->head = qdata;
-  if (q->tail)
-    q->tail->next = qdata;
-  q->tail = qdata;
-  q->length++;
+  if (q->max_data > 0U && q->length >= q->max_data) {
+    nns_edge_logw ("[Queue] Cannot push new data, max data in queue is %u.",
+        q->max_data);
```

Comment:
How about setting the buffer to drop out of the old or new buffer in the next patch?

Reply:
Yes. It's ok to have a backlog for a leaky queue.

Reply:
I will add new PR for leaky option.